numpy/lib/function_base.py

```diff
@@ -1299,7 +1299,6 @@ def piecewise(x, condlist, funclist, *args, **kw):
                                    isinstance(condlist[0], ndarray))):
         condlist = [condlist]
     condlist = array(condlist, dtype=bool)
-    n = len(condlist)
     # This is a hack to work around problems with NumPy's
     #  handling of 0-d arrays and boolean indexing with
     #  numpy.bool_ scalars
```

Comment:
if the hack lines1306--1311 is removed, this function would not work for the case that 'x' is scholar. So, I consider that the hack will not able to be removed.
The problem is that the condlist is updated by the hack, but the length is not updated.